Feature/course media model

RealBreadBoi opened the pull request:
Added course media model

RealBreadBoi committed:
Auto stash before merge of "master" and "origin/master"

Added model for course media + flyway migration

RealBreadBoi committed:
Minor fix

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/models/Media.java:
```diff
@@ -0,0 +1,24 @@
+
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "media_type", discriminatorType = DiscriminatorType.INTEGER)
```
For better readability it would be bettern if the disciminator type will be string and it will contain the value from the enum such that we won't be checking back and forth which is which as a type

RealBreadBoi replied:
done

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/models/Media.java:
```diff
@@ -0,0 +1,24 @@
+public abstract class Media extends BaseEntity {
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    private Course course;
```
Please also map the media on the course so we can retrieve it more easily.

teodorvoicencu reviewed (commented):
Also i think it would be better to attribute the media to the lesson entity rather than course.

RealBreadBoi committed:
Made mapping between course and media bidirectional. Added orderIndex to Media.

RealBreadBoi committed:
Changed discriminator column to string

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/models/Course.java:
```diff
@@ -44,4 +43,7 @@
     private User ownerId;
+
+    @OneToMany(mappedBy = "course")
+    private List<Media> media;
```
> When you need a set and never care about order, use a Set. When for some reason order is important (ordered list, ordering by date, etc.), then a List.

In our case make sure the media is returned by repository ordered by orderIndex, otherwise use a set.

teodorvoicencu commented on backend/src/main/resources/db/migration/V07__create_media.sql:
```diff
@@ -0,0 +1,10 @@
+    text varchar(1025),
+    download_link varchar(1025),
+    youtube_link varchar(1025),
+    image_link varchar(1025),
```
255 i think should be enough

teodorvoicencu commented on backend/src/main/resources/db/migration/V07__create_media.sql:
```diff
@@ -0,0 +1,10 @@
+    ID serial primary key,
+    course_id integer references courses(ID),
+    order_index integer,
+    text varchar(1025),
```
there is a text data type in postgres sql, use that and change this field name. Something like `content text` maybe

teodorvoicencu commented on backend/src/main/java/ubb/courses/backend/repositories/MediaRepository.java:
```diff
@@ -0,0 +1,10 @@
+import java.util.List;
+
+public interface MediaRepository extends JpaRepository<Media, Integer> {
+    List<Media> findMediaByCourse_Id(Integer courseId);
```
Use set if we do not want to have an implied order because we have orderIndex. Or use list and make sure it is always ordered by the orderIndex.

teodorvoicencu committed:
Change media into lesson

teodorvoicencu committed:
Code refactor
